### PR TITLE
Adding webpack module build support

### PR DESCRIPTION
**Pull request by sachingupta**
This change will add webpack support in teams-js code to support
modules (Currently we have one js file where we add all interfaces and code).

Followup change will be to break existing JS file into different module files.

To test please try
Option 1. npm package
https://www.npmjs.com/package/@microsoft/teams-js/v/1.4.0-beta.3.9
import { microsoftTeams } from '@microsoft/teams-js'

2. Option 2: as standalone JS
 <script src="https://unpkg.com/@microsoft/teams-js@1.4.0-beta.3.9/dist/MicrosoftTeams.min.js"></script>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,47 +1,50 @@
 {
   "name": "@microsoft/teams-js",
   "author": "Microsoft Teams",
-  "version": "1.3.6",
+  "version": "1.4.0-beta.3.9",
```
Comment (sachingupta) [inline package.json]: will revert this

File: src/MicrosoftTeams.ts [hunk 2]
```diff
@@ -485,11 +488,11 @@ namespace microsoftTeams {
   let initializeCalled = false;
 
   let isFramelessWindow = false;
-  let currentWindow: Window;
-  let parentWindow: Window;
+  let currentWindow: any | any;
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Shouldn't this be `Window | any`?

File: src/MicrosoftTeams.ts [hunk 3]
```diff
@@ -485,11 +488,11 @@ namespace microsoftTeams {
...
+  let parentWindow: Window | any;
   let parentOrigin: string;
   let parentMessageQueue: MessageRequest[] = [];
-  let childWindow: Window;
+  let childWindow: Window | any;
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Why do you need to change the type here (and above)? In what scenario does this need to be `any`?

File: src/MicrosoftTeams.ts [hunk 4]
```diff
@@ -830,6 +838,29 @@ namespace microsoftTeams {
     sendMessageRequest(parentWindow, "openFilePreview", params);
   }
 
+  export interface ShowNotificationParameters {
+    message: string;
+    isDownloadComplete: boolean;
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: I assume you will change this to an enum, right?

File: src/MicrosoftTeams.ts [hunk 5]
```diff
@@ -830,6 +838,29 @@ namespace microsoftTeams {
...
+  }
+
+  /**
+   * @private
+   * Hide from docs.
+   * ------
+   * download file.
+   * @param file The file to download.
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Update the comments once you make this API more generic.

File: tsconfig.json [hunk 6]
```diff
@@ -1,15 +1,20 @@
 {
   "compilerOptions": {
+    "outDir": "./dist",
+    "declarationDir": "./dts",
+    "target": "es6",
```
Comment (ydogandjiev) [inline tsconfig.json]: Hmm... Shouldn't the target be es5? Some of the browsers we need to support can't handle es6 yet.

File: src/MicrosoftTeams.ts (hunk 5 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: yes

File: src/MicrosoftTeams.ts (hunk 4 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: yes

File: src/MicrosoftTeams.ts (hunk 3 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: Looks like we are adding new Window interface that override WIndow interface. will fix it

**Pull request merged**